samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -538,9 +544,8 @@ def _resolve_resource_attribute(
 
     attribute_value = resource.attributes.get(attribute_name)
     if attribute_value is None:
```

Comment:
if the value of the attribute is None, it means that the customer does not set value for it in the terraform application, and so we should not raise an error from this function. The function caller if it expect a value for this attribute, so it should raise the error

Reply:
What does `The function caller if it expect a value for this attribute, so it should raise the error` mean? we raise the error higher up in the chain?

Reply:
Yes, in this function, we will not raise an error for this scenario in this function (this is a utility function). So, for example if we used this utility function in a scenario that we expect the resource contains a a specific attribute, and we received it as None, so we should raise this error from the caller function and not the utility function